False positive for D417 (Part 2)

Opening a new issue here to track the follow-on comment I left here:

https://github.com/charliermarsh/ruff/issues/792#issuecomment-1347239427

Transcribing that comment:

Thanks for the fix for #792! However, I've noticed that the error returns if there is more than one argument present:
```python
# test.py
def f(x, y, z):
    """Do f.

    Args:
        x: the value
        y: Another thing
        z: A final argument

    Returns: the value
    """
    return x
```

$ ruff --version
ruff 0.0.177

$ ruff --select D test.py
Found 1 error(s).
test.py:1:1: D417 Missing argument descriptions in the docstring: `y`, `z`

The error again goes away if the Returns block gets a newline. Insisting on a newline there is actually fine for me, but the confusing error message is still unfortunate.

--- a/src/pydocstyle/plugins.rs
+++ b/src/pydocstyle/plugins.rs
@@ -1349,39 +1349,22 @@ fn missing_args(checker: &mut Checker, docstring: &Docstring, docstrings_args: &
 
 // See: `GOOGLE_ARGS_REGEX` in `pydocstyle/checker.py`.
 static GOOGLE_ARGS_REGEX: Lazy<Regex> =
-    Lazy::new(|| Regex::new(r"^\s*(\w+)\s*(\(.*?\))?\s*:\n?\s*.+").unwrap());
+    Lazy::new(|| Regex::new(r"^\s*(\w+)\s*(\(.*?\))?\s*:.+").unwrap());
 
 fn args_section(checker: &mut Checker, docstring: &Docstring, context: &SectionContext) {
-    let mut args_sections: Vec<String> = vec![];
-    for line in textwrap::dedent(&context.following_lines.join("\n"))
-        .trim()
-        .lines()
-    {
-        if line.chars().next().map_or(true, char::is_whitespace) {
-            // This is a continuation of documentation for the last
-            // parameter because it does start with whitespace.
-            if let Some(current) = args_sections.last_mut() {
-                current.push_str(line);
-            }
-        } else {
-            // This line is the start of documentation for the next
-            // parameter because it doesn't start with any whitespace.
-            args_sections.push(line.to_string());
+    let mut matches = Vec::new();
+    for line in context.following_lines {
+        if let Some(captures) = GOOGLE_ARGS_REGEX.captures(line) {
+            matches.push(captures);
         }
     }
 
     missing_args(
         checker,
         docstring,
-        // Collect the list of arguments documented in the docstring.
-        &args_sections
+        &matches
             .iter()
-            .filter_map(
-                |section| match GOOGLE_ARGS_REGEX.captures(section.as_str()) {
-                    Some(caps) => caps.get(1).map(|arg_name| arg_name.as_str()),
-                    None => None,
-                },
-            )
+            .filter_map(|captures| captures.get(1).map(|arg_name| arg_name.as_str()))
             .collect(),
     );
 }

